### PR TITLE
Add cluster resource

### PR DESCRIPTION
**Pull request by CarlosRoGuerra**

### CODE REVIEW

File: tsuru/resource_tsuru_cluster.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	return &schema.Resource{
+		CreateContext: resourceTsuruClusterCreate,
+		ReadContext:   resourceTsuruClusterRead,
+		UpdateContext: resourceTsuruClusterUpdate,
+		DeleteContext: resourceTsuruClusterDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
```
Comment (pedrokiefer) [inline tsuru/resource_tsuru_cluster.go:21]: We should add `Description` to all schema items.

File: tsuru/resource_tsuru_cluster.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+				},
+				Optional: true,
+			},
+			"tsuru_provisioner": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "kubernetes",
+			},
+			"ca_cert": {
```
Comment (pedrokiefer) [inline tsuru/resource_tsuru_cluster.go:40]: Cluster all certificate related fields for easier readability.

Comment (pedrokiefer) [reply tsuru/resource_tsuru_cluster.go:40]: Maybe more verbose names, such as `ca_certificate`, `client_certificate`...

Comment (wpjunior) [reply tsuru/resource_tsuru_cluster.go:40]: @pedrokiefer I suggested this field to be compliant to tsuru types package.

File: tsuru/resource_tsuru_cluster.go (hunk 1 above)
Comment (wpjunior) [reply tsuru/resource_tsuru_cluster.go:21]: yeap @pedrokiefer, let's add after merge

File: tsuru/resource_tsuru_cluster.go (hunk 2 above)
Comment (pedrokiefer) [reply tsuru/resource_tsuru_cluster.go:40]: My take is that terraform should be easier to understand than the API. Keeping it equal to the types package you are exposing the API to another level of abstraction -- the provider should abstract this for our users.

File: tsuru/resource_tsuru_cluster.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+	name := d.Get("name").(string)
+
+	cluster := tsuru.Cluster{
+		Name:        name,
+		Addresses:   addresses,
+		Provisioner: d.Get("tsuru_provisioner").(string),
+		Cacert:      []byte(d.Get("ca_cert").(string)),
+		Clientcert:  []byte(d.Get("client_cert").(string)),
+		Clientkey:   []byte(d.Get("client_key").(string)),
+		Pools:       []string{},
```
Comment (arthurcgc) [inline tsuru/resource_tsuru_cluster.go:88]: Are we going to leave this hardcoded? shouldn't we add []Pools in schema?

File: tsuru/resource_tsuru_cluster_test.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	echo "github.com/labstack/echo/v4"
+	"github.com/stretchr/testify/require"
+	"github.com/tsuru/go-tsuruclient/pkg/tsuru"
+)
+
+func TestAccTsuruCluster_basic(t *testing.T) {
+	fakeServer := echo.New()
+	fakeServer.POST("1.3/provisioner/clusters", func(c echo.Context) error {
+		// p := &tsuru.Cluster{}
```
Comment (arthurcgc) [inline tsuru/resource_tsuru_cluster_test.go:19]: commented code :/

Comment (arthurcgc) [reply tsuru/resource_tsuru_cluster_test.go:19]: I think we should have at least two basic tests: one with the least number of arguments passed, going through the default values and one using only custom values to create the resource...

File: tsuru/resource_tsuru_cluster_test.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+		IDRefreshName:     resourceName,
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      nil,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTsuruClusterConfig_basic(server.URL, "test_cluster"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccResourceExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "custom_data.token", "test_token"),
+					//resource.TestCheckResourceAttr("tsuru_cluster.test_cluster", "tsuru_provisioner", "kubernetes"),
```
Comment (arthurcgc) [inline tsuru/resource_tsuru_cluster_test.go:61]: commented code and we need to check all resource attributes here as well...

**Pull request merged**